platform/emdrv/nvm3/src/ccp_flash_intf.c

```diff
@@ -0,0 +1,349 @@
+#define M4SS_QSPI_D2 56
+#define M4SS_QSPI_D3 57
+#endif
+
+void RSI_QSPI_PinMuxInit(void) {
+#ifdef CHIP_9118
+  /*Pad selection enable */
+  RSI_EGPIO_PadSelectionEnable(15);
+#endif
+
+#ifdef CHIP_9117
+  /*Pad selection enable */
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_CLK);
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_D0);
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_D1);
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_CSN0);
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_D2);
+  RSI_EGPIO_PadSelectionEnable(PadSelectionEnable_D3);
+#endif
+
+  /*Receive enable for QSPI GPIO*/
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_CLK);
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_CSN0);
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_D0);
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_D1);
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_D2);
+  RSI_EGPIO_PadReceiverEnable(M4SS_QSPI_D3);
+
+  /*Set GPIO pin MUX for QSPI*/
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_CLK, QSPI_MODE);
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_CSN0, QSPI_MODE);
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_D0, QSPI_MODE);
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_D1, QSPI_MODE);
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_D2, QSPI_MODE);
+  RSI_EGPIO_SetPinMux(EGPIO, 0, M4SS_QSPI_D3, QSPI_MODE);
+}
+
+/* QSPI HAL wrapper for NVM3 */
+uint32_t RSI_FLASH_Initialize(void) {
+  /*Init the QSPI configurations structure */
+  spi_config_t spi_configs_init;
+
+  /*Get QSPI  structures configuration     */
+  GetQspiConfig(&spi_configs_init);
+
+  /* Configures the pin MUX for QSPI  pins*/
+  RSI_QSPI_PinMuxInit();
+
+  /* initializes QSPI  */
+  RSI_QSPI_SpiInit((qspi_reg_t *)QSPI_BASE, &spi_configs_init, 1, 0, 0);
+
+  return 0;
+}
+
+uint32_t RSI_FLASH_EraseSector(uint32_t sector_address) {
+  spi_config_t spi_configs_erase;
+  GetQspiConfig(&spi_configs_erase);
+  // ROMAPI_QSPI_API->qspi_spi_erase((qspi_reg_t
+  // *)(M4_QSPI_BASE_ADDRESS),&spi_configs_erase,SECTOR_ERASE, sector_address, 1,
+  // 0);
+  /* The erase function does the erase and takes care of any missing
+   * configurations to successfully erase the sector*/
+  // return 0;
+
+  /* Erases the SECTOR   */
+  DEBUGOUT("\r\n Erase Sector \r\n");
+  RSI_QSPI_SpiErase((qspi_reg_t *)QSPI_BASE, &spi_configs_erase, SECTOR_ERASE,
+                    sector_address, 1, 0);
+
+  /* The erase function does the erase and takes care of any missing
+   * configurations to successfully erase the sector*/
+  return 0;
+}
+
+uint32_t RSI_FLASH_Write(uint32_t address, unsigned char *data,
+                         uint32_t length) {
+  spi_config_t spi_configs_program;
+  uint32_t check_sum = 0;
+  GetQspiConfig(&spi_configs_program);
+
+  /* writes the data to required address using qspi */
+  DEBUGOUT("\r\n Write Data to Flash Memory \r\n");
+  RSI_QSPI_SpiWrite((qspi_reg_t *)QSPI_BASE, &spi_configs_program, 0x2, address,
+                    (uint8_t *)&data[0], length,
+                    256, // 256
```

Comment:
What is this 256? block size? max write? It might be valuable to create a define with a relevant name